test/apply.lean

```diff
@@ -7,9 +7,9 @@ example : ∀ n m : ℕ, n + m = m + n :=
 begin
   apply' nat.rec,
   -- refine nat.rec _ _,
+  { intro m, ring },
```

Comment:
What is this about? How did this proof break?

Reply:
See #3777